core/src/test/java/com/redhat/lightblue/client/request/TestDataInsertRequest.java

```diff
@@ -31,4 +32,14 @@ public void testProjectionsAsArray() {
                 "\"projection\":{\"field\":\"*\",\"include\":true,\"recursive\":false}"));
     }
 
+    @Test
+    public void testExecutionReadPreference() {
```

Comment:
Only read preference option is tested, but writeConcern and maxQueryTimeMS were included in implementation.